Source/CsDebugScript.Engine/Engine/Utility/SimpleCache.cs

```diff
@@ -1,8 +1,44 @@
 ﻿using System;
+using System.Collections;
 
 namespace CsDebugScript.Engine.Utility
 {
     /// <summary>
+    /// Interface for all caching structures.
+    /// </summary>
+    public interface ICache
+    {
+        /// <summary>
+        /// Gets a value indicating whether value is cached.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if cached; otherwise, <c>false</c>.
+        /// </value>
+        bool Cached { get; }
+
+        /// <summary>
+        /// Invalidates this cache.
+        /// </summary>
+        void InvalidateCache();
+
+        /// <summary>
+        /// Gets or sets the value. The value will be populated if it wasn't cached.
+        /// </summary>
+        object ValueRaw { get; }
+    }
+
+    /// <summary>
+    /// Interface for all caching collections.
+    /// </summary>
+    public interface ICacheCollection : ICache
+    {
+        /// <summary>
+        /// Returns IEnumerable for all cached values.
+        /// </summary>
+        IEnumerable ValuesRaw { get; }
```

Comment:
Yes, implementing IEnumerable makes things much cleaner.

---
In reply to: [70603041](https://github.com/southpolenator/WinDbgCs/pull/8#discussion_r70603041) [](ancestors = 70603041,70602441)